### PR TITLE
Tick blocks in cubes

### PR DESCRIPTION
**Pull request by Foghrye4**
Seems to fix "cube saving console flood" issue. Also works fine.

### CODE REVIEW

File: src/main/java/cubicchunks/util/XYZPseudoFractalMap.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
+package cubicchunks.util;
```
Comment (Foghrye4) [inline src/main/java/cubicchunks/util/XYZPseudoFractalMap.java]: Dunno why its still here. I thought I stash it in another branch.

**Comment by Niko-sk2x** [comment]
We should keep it like in vanilla - only 1000 block ticks per server tick. Otherwise the first server tick after world generation will take a long time. I think the best solution would be to just have all the data dumplicated in cubes, and let World do the ticking. Then you would just need a few injections in WorldServer to keep the values in Cubes and in World in sync.

**Comment by Foghrye4** [comment]
> have all the data duplicated in cubes, and let World do the ticking.

I don't understand you. Why we should duplicate data in cubes if we don't want to use it?
Also vanilla updates limit per tick is 65536. See WorldServer Line 743.

I added a limit of 1000 ticks per cube.

**Comment by Niko-sk2x** [comment]
Oh, it used to be 1000 I think. Maybe they changed it. We would use the data when saving cubes. Instead of filtering, we would use data directly from cubes.

**Comment by Foghrye4** [comment]
I don't see anywhere for 'pendingTickListEntriesHashSet' or 'pendingTickListEntriesTreeSet' is accessed for saving this data. Does that data really saved on hard drive?
Also a real goal of this PR is to prevent block updates inside cubes which are not fully populated. To do so outside cube instance we still need to filter those updates.

**Comment by Niko-sk2x** [comment]
Running block ticks in not-fully-populated areas is ok, because the only situations where these ticks occur is when player places blocks, and in already populated areas. One major case is falling gravel. It doesn't really matter that much when these blocks fall while the cube they are in isn't completely populated, because the place they are in already is populated.

As for where they are accessed - CubicChunks accesses them via reflection, vanilla has a special method that filters them for StructureBoundingBox, but it only uses X and Z coordinates.

And the limit you added also isn't per-world limit. It also may do the updates in the wrong order.

**Comment by Foghrye4** [comment]
@Barteks2x so I should add a saving mechanism of 'pendingTickListEntriesTreeSet' on cube saving event and this PR will be OK?

File: src/main/java/cubicchunks/server/CubeProviderServer.java [hunk 2]
```diff
@@ -187,6 +188,11 @@ public boolean saveChunks(boolean alwaysTrue) {
     public boolean tick() {
         // NOTE: the return value is completely ignored
         // NO-OP, This is called by WorldServer's tick() method every tick
```
Comment (Cyclonit) [inline src/main/java/cubicchunks/server/CubeProviderServer.java]: This does no longer look like a no-op to me.

Comment (Niko-sk2x) [reply src/main/java/cubicchunks/server/CubeProviderServer.java]: You should probably change that. Then it should be ok to merge for now. If I want any further changes, I can do it myself some time later.

**Review by Niko-sk2x — CHANGES_REQUESTED** [review]

**Comment by Foghrye4** [comment]
Requested done.

**Review by Niko-sk2x — APPROVED** [review]

**Pull request merged**